hmmlearn/hmm.py

```diff
@@ -520,134 +520,221 @@ class GMMHMM(_BaseHMM):
     transmat\_ : array, shape (n_components, n_components)
         Matrix of transition probabilities between states.
 
-    gmms\_ : list of GMM objects, length n_components
-        GMM emission distributions for each state.
+    means\_ : array, shape (n_components, n_mix)
+
+    covars\_ : array
+        Covariance parameters for each state in each component.
+
+        The shape depends on ``covariance_type``::
+
+            (n_components, n_mix)                          if 'spherical',
+            (n_components, n_features, n_features)         if 'tied',
+            (n_components, n_mix, n_features)              if 'diag',
+            (n_components, n_mix, n_features, n_features)  if 'full'
 
-    Examples
-    --------
-    >>> from hmmlearn.hmm import GMMHMM
-    >>> GMMHMM(n_components=2, n_mix=10, covariance_type='diag')
-    ... # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
-    GMMHMM(algorithm='viterbi', covariance_type='diag',...
     """
 
     def __init__(self, n_components=1, n_mix=1,
                  startprob_prior=1.0, transmat_prior=1.0,
-                 covariance_type='diag', covars_prior=1e-2,
-                 algorithm="viterbi", random_state=None,
-                 n_iter=10, tol=1e-2, verbose=False,
-                 params="stmcw", init_params="stmcw"):
+                 algorithm='viterbi', covariance_type='full',
+                 random_state=None, n_iter=10, tol=1e-2,
+                 verbose=False, params='stmcw',
+                 init_params='stmcw'):
         _BaseHMM.__init__(self, n_components,
                           startprob_prior=startprob_prior,
                           transmat_prior=transmat_prior,
                           algorithm=algorithm, random_state=random_state,
                           n_iter=n_iter, tol=tol, verbose=verbose,
                           params=params, init_params=init_params)
-
-        if covariance_type != "diag":
-            warnings.warn("Fitting a GMMHMM with {0!r} covariance type "
-                          "is broken in 0.2.0. Please update to 0.2.1 once "
-                          "it's available.".format(covariance_type),
-                          UserWarning)
-
-        # XXX: Hotfit for n_mix that is incompatible with the scikit's
-        # BaseEstimator API
-        self.n_mix = n_mix
         self.covariance_type = covariance_type
-        self.covars_prior = covars_prior
-        self.gmms_ = []
-        for x in range(self.n_components):
-            if covariance_type is None:
-                gmm = GMM(n_mix, random_state=self.random_state)
-            else:
-                gmm = GMM(n_mix, covariance_type=covariance_type,
-                        random_state=self.random_state)
-            self.gmms_.append(gmm)
+        self.n_mix = n_mix
 
     def _init(self, X, lengths=None):
-        super(GMMHMM, self)._init(X, lengths=lengths)
+        super(GMMHMM, self)._init(X, lengths)
```

Comment:
Please pass `length` as a keyword argument. It's part of the internal API contract.